cmd/entrypoints/clusterresource.go

```diff
@@ -3,32 +3,50 @@ package entrypoints
 import (
 	"context"
 
+	"github.com/flyteorg/flyteadmin/pkg/executioncluster/interfaces"
+
+	"github.com/flyteorg/flytestdlib/promutils"
+
+	"github.com/flyteorg/flyteidl/clients/go/admin"
+
 	"github.com/flyteorg/flyteadmin/pkg/clusterresource"
+	"github.com/flyteorg/flyteadmin/pkg/config"
 	executioncluster "github.com/flyteorg/flyteadmin/pkg/executioncluster/impl"
 	"github.com/flyteorg/flyteadmin/pkg/runtime"
+	runtimeInterfaces "github.com/flyteorg/flyteadmin/pkg/runtime/interfaces"
 	"github.com/flyteorg/flytestdlib/logger"
 
-	"github.com/flyteorg/flyteadmin/pkg/config"
-	"github.com/flyteorg/flyteadmin/pkg/repositories"
-	repositoryConfig "github.com/flyteorg/flyteadmin/pkg/repositories/config"
-	"github.com/flyteorg/flytestdlib/promutils"
 	"github.com/spf13/cobra"
 	_ "gorm.io/driver/postgres" // Required to import database driver.
-	gormLogger "gorm.io/gorm/logger"
 )
 
 var parentClusterResourceCmd = &cobra.Command{
 	Use:   "clusterresource",
 	Short: "This command administers the ClusterResourceController. Please choose a subcommand.",
 }
 
-func GetLocalDbConfig() repositoryConfig.DbConfig {
-	return repositoryConfig.DbConfig{
-		Host:   "localhost",
-		Port:   5432,
-		DbName: "postgres",
-		User:   "postgres",
+func getClusterResourceController(ctx context.Context, scope promutils.Scope, configuration runtimeInterfaces.Configuration) clusterresource.Controller {
+	initializationErrorCounter := scope.MustNewCounter(
+		"flyteclient_initialization_error",
+		"count of errors encountered initializing a flyte client from kube config")
+	var listTargetsProvider interfaces.ListTargetsInterface
+	var err error
+	if len(configuration.ClusterConfiguration().GetClusterConfigs()) == 0 {
+		serverConfig := config.GetConfig()
+		listTargetsProvider, err = executioncluster.NewInCluster(initializationErrorCounter, serverConfig.KubeConfig, serverConfig.Master)
+	} else {
+		listTargetsProvider, err = executioncluster.NewListTargets(initializationErrorCounter, executioncluster.NewExecutionTargetProvider(), configuration.ClusterConfiguration())
+	}
+	if err != nil {
+		panic(err)
+	}
+
+	clientSet, err := admin.ClientSetBuilder().WithConfig(admin.GetConfig(ctx)).Build(ctx)
+	if err != nil {
+		panic(err)
 	}
+
+	return clusterresource.NewClusterResourceController(clientSet.AdminClient(), listTargetsProvider, scope)
```

Comment:
Will it crash if it can't talk to admin? just making sure it'll... specially for enabling auth in admin, this needs to know to detect that and crash...

Reply:
it will dump a bunch of error logs complaining admin is unreachable and fail